app/assets/javascripts/components/story/StoryAttachment.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+class StoryAttachment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.saveInput = (input) => { this.filesInput = input };
+    this.state = {
+      random: (Math.floor(Math.random() * 10000) + 1),
+      progressElementId: '',
+      finishedElementId: '',
+      attachinaryContainerId: ''
+    }
+  }
+
+  componentWillMount() {
+    this.setState({
```

Comment:
Why do we need that in the state if it never changes? You can set it directly on the instance inside the constructor, not necessary to use `componentWillMount`.